### PR TITLE
feat(workers): allow to ignore the last X minor or major releases

### PR DESCRIPTION
**Pull request by quentinus95**
## Changes

This pull request provides two new configuration options, please read the context section to understand why this is useful:
- `ignoreLastXMajor`: ignore the last X major versions 
- `ignoreLastXMinor`: ignore the last X minor versions

## Context

The aim of this pull request is to help to ignore newer versions of a dependency or package that might provide a decrease of stability.

This is particularly useful when having dependencies that:
- release a lot of new versions in a short amount of time (e.g., an NPM dependency that release a patch every week, and a minor every month)
- have monthly releases (e.g., GitLab releases a new version every month. It can be useful to upgrade to the last patch of the previous release every month instead of the new minor right away)
- are critical dependencies (e.g., always ignoring the last two major releases of Kubernetes while benefiting from the last patches of the previous majors).

Today, the number of updates can be overwhelming. Having a way to ignore most of the consecutive patches and always update to the last patch of the previous minor can provide a good trade-off between last features and quick fixes.

Packages sometimes provide maintenance rules for multiple versions at the same time (e.g., Symfony might release a patch for critical security issues over multiple majors). With this new configuration solution you can still benefit from those fixes. You may also sometimes need to plan for extensive work when upgrading to a major version. For such situations, Renovate might not be very useful. You know that you need to update to a major, it will require several weeks for your team to upgrade. You don't want to be flooded by Renovate for each patch on this version.

Finally, you can combine both options. For instance, you may want to ignore the last major and the last minor of a specific package (e.g., Kubernetes).

## Documentation (please check one with an [x])

I need help on what and where I should update documentation. Moreover, I need help to understand how I can plug my feature to the actual configuration options.

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Please create a github issue to discuss requirements/naming/etc first, you can copy/paste some of the context from this PR

**Comment by quentinus95** [comment]
@rarkins done, see https://github.com/renovatebot/renovate/issues/18236 :pray:

**Review by viceice — COMMENTED** [review]
still needs discussion, see issue

File: lib/workers/repository/process/lookup/filter.ts [hunk 1]
```diff
@@ -41,6 +41,42 @@ export function filterVersions(
...
+  // If configuration uses ignoreLastX..., exclude versions that are too recent
+  const { ignoreLastXMajor, ignoreLastXMinor } = config;
+
+  if (ignoreLastXMajor) {
+    const lastMajor = Math.max(
+      ...filteredVersions.map(
+        (release) => versioning.getMajor(release.version)!
+      )
+    );
+    if (lastMajor !== null) {
```
Comment (rarkins) [inline lib/workers/repository/process/lookup/filter.ts:53]:
```suggestion
    if (is.number(lastMajor)) {
```

File: lib/workers/repository/process/lookup/filter.ts [hunk 2]
```diff
@@ -41,6 +41,42 @@ export function filterVersions(
...
+      ...filteredVersions.map(
+        (release) => versioning.getMajor(release.version)!
+      )
+    );
+    const lastMinor = Math.max(
+      ...filteredVersions
+        .filter((release) => versioning.getMajor(release.version) === lastMajor)
+        .map((release) => versioning.getMinor(release.version)!)
+    );
+    if (lastMinor !== null) {
```
Comment (rarkins) [inline lib/workers/repository/process/lookup/filter.ts:71]:
```suggestion
    if (is.number(lastMinor)) {
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Pull request closed without merging**